dlp/src/test/java/dlp/snippets/InspectTests.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dlp.snippets;
+
+import static junit.framework.TestCase.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class InspectTests {
+
+  private ByteArrayOutputStream bout;
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(
+        System.getenv(varName),
+        "Environment variable '%s' is required to perform these tests.".format(varName)
+    );
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void beforeTest() {
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(null);
+    bout.reset();
+  }
+
+  @Test
+  public void testInspectString() {
+    InspectString.inspectString(PROJECT_ID,  "I'm Gary and my email is gary@example.com");
+
+    String output = bout.toString();
+    assertThat(output, CoreMatchers.containsString("Info type: PHONE_NUMBER"));
```

Comment:
Wait, is this verifying that the DLP API found a PHONE_NUMBER in that test string? Because I don't see one. If the output is only verifying that it was _looking_ for phone numbers, then verifying that it was also looking for email addresses doesn't verify that it found one...

Reply:
The snippet prints out any findings out to console, so this test verifies that at least one of the findings had the infotype `PHONE_NUMBER`. 

The test was incorrect to look for an `"EMAIL_ADDRESS"` - I had changed the test string but forgot to update the following asserts.